mono/mini/mini-runtime.c

```diff
@@ -3730,7 +3730,7 @@ mini_init (const char *filename, const char *runtime_version)
 	mono_thread_attach (domain);
 #endif
 
-	if (mono_profiler_events & MONO_PROFILE_STATISTICAL)
+	if (mono_profiler_enabled ())
```

Comment:
Yeah, this is a really good argument for why we should not be calling `mono_profiler_enable` in `mono_profiler_new`/`mono_profiler_install`, only in `mono_profiler_load`.